tests/js/server/aql/aql-functions-date.js

```diff
@@ -2427,6 +2427,36 @@ function ahuacatlDateFunctionsTestSuite () {
         assertEqual([ value[1] ], getQueryResults("RETURN DATE_ISO8601(@value)", { value: value[0] }));
       });
     },
+	
+	// //////////////////////////////////////////////////////////////////////////////
+    // / @brief test date_utctolocal and date_localtoutc functions
+    // //////////////////////////////////////////////////////////////////////////////
+
+    testDateUtcToLocalAndLocalToUtc: function () {
+      var values = [
+        [ "2020-03-15T01:00:00.000Z", "Europe/Berlin", "2000-03-15T00:00:00.000Z"],
+		[ "2020-04-15T02:00:00.000Z", "Europe/Berlin", "2000-04-15T00:00:00.000Z"],
+		[ "2020-10-14T21:00:00.999Z", "America/New_York", "2020-10-15T01:00:00.999Z"],
+		[ "2020-11-14T20:00:00.999Z", "America/New_York", "2020-11-15T01:00:00.999Z"],
+		[ "1991-09-10T09:00:00.999Z", "Asia/Shanghai", "1991-09-10T00:00:00.999Z"],
+		[ "1991-09-20T08:00:00.999Z", "Asia/Shanghai", "1991-09-20T00:00:00.999Z"],
+		[ "1992-09-10T08:00:00.999Z", "Asia/Shanghai", "1992-09-10T00:00:00.999Z"],
+		[ "1992-09-20T08:00:00.999Z", "Asia/Shanghai", "1992-09-20T00:00:00.999Z"],		
+		[ "2020-07-07T09:30:00.333Z", "Australia/Darwin", "2020-07-07T00:00:00.333Z" ],
+		[ "2020-07-07T08:45:00.333Z", "Australia/Eucla", " 2020-07-07T00:00:00.333Z" ],
+		[ "2020-04-15T06:06:06.111Z", "UTC", "2020-04-15T06:06:06.111Z" ],
+		[ "2020-07-07T08:45:00.333Z", "Etc/GMT+6", "2020-07-06T18:00:00.333Z" ],
+		[ "2020-07-07T08:45:00.333Z", "Etc/GMT-6", "2020-07-07T06:00:00.333Z" ]
```

Comment:
The "Z" suffix is used to identify UTC times (see https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators).

The converted local times should either have no suffix, or one that contains the time offset. E.g., `2020-03-15T01:00:00.000+01:00

Reply:
Yeah I know, I was still hoping to get away with it ;-p 
because I don't know yet how all other date functions will behave

aqlTimeValue hardcodes this as it assumens all times to to be utc (system time)

I will make an alternative implementation that accepts zoned_time

from there we can discuss what  to return from aql.

2020-07-06T18:00:00.333

do an overload with format=true
2020-07-06T18:00:00.333 CET

Reply:
so dateParse does infact apply TZD (as it should)
for me that means I will just omit Z if offset or saving <> 0

This raises the question however should DATE_ADD convert to system time before calculation?
2020-02-01T00:00:00.000+01:00 + 1 Month wont work with that you need to keep local time

thus rewriting all DATE_ functions with zoned_time support...

Reply:
Test looks good now. The question about zoned_time support for all DATE_* functions is something the arango team has to discuss, but this is probably something that could be tackled in a separate PR.

Reply:
Yeah maybe something for 4.0 as this will break things, now atleast the groundwork for all that is in atleast.